setup.cfg

```diff
@@ -152,6 +152,8 @@ filterwarnings =
     ignore:parallel reading does not currently work, so falling back to serial
     # Ignore deprecation warning triggered by s3fs (fsspec/s3fs#716)
     ignore:'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning
+    # Ignore numpy 2.0 deprecation warning triggered by pytest
+    ignore:`maximum_sctype` is deprecated:DeprecationWarning
```

Comment:
Is this the warning from scipy dev?

Reply:
I don't think we should ignore it if we can help it if it can be resolved by new scipy dev wheel soon.

Reply:
It is from `pytest`. I had to insert this otherwise I could not run a single test! But I agree it should go once `pytest` is updated.

Reply:
Oh wait, the scipy one is `E   ImportError: cannot import name 'ComplexWarning' from 'numpy' `, not this one. Where did `maximum_sctype` come from?

Reply:
Since when pytest depends on numpy? This doesn't seem right. Can we remove this warning so I can see the traceback? Thanks!

Reply:
It looks like we use `np.maximum_sctype` in `nddata/bitmask.py`.  I'm running into this too with `devdeps` for `photutils` tests

Reply:
Here's the traceback without it -- you are right it is not due to pytest. Will fix...
```
mhvk@weasel:~/data/mhvk/packages/astropy$ source .tox/py311-test-devdeps/bin/activate
(py311-test-devdeps) mhvk@weasel:~/data/mhvk/packages/astropy$ git checkout upstream/main setup.cfg
Updated 1 path from 5459035554
(py311-test-devdeps) mhvk@weasel:~/data/mhvk/packages/astropy$ pytest
=================================================== test session starts ===================================================
platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0

Running tests with Astropy version 6.0.dev437+g12e5d131bd.
Running tests in astropy docs.

Date: 2023-08-11T18:02:45

Platform: Linux-6.3.0-1-amd64-x86_64-with-glibc2.37

Executable: /home/mhvk/data/mhvk/packages/astropy/.tox/py311-test-devdeps/bin/python

Full Python Version: 
3.11.4 (main, Jun  7 2023, 10:13:09) [GCC 12.2.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 2.0.0.dev0+778.ga2bfa54f8
Scipy: 1.12.0.dev0+1440.64be173
Matplotlib: 3.8.0rc1
h5py: not available
Pandas: not available
PyERFA: 2.0.0.4.dev3+gc12b70e
Cython: not available
Scikit-image: not available
asdf-astropy: 0.4.1.dev34+g0ab96f6
pyarrow: not available

Using Astropy options: remote_data: none.

CI: undefined
ARCH_ON_CI: undefined
IS_CRON: undefined

rootdir: /home/mhvk/data/mhvk/packages/astropy
configfile: setup.cfg
testpaths: astropy, docs
plugins: arraydiff-0.5.0, astropy-header-0.2.2, mock-3.11.1, filter-subpackage-0.1.2, doctestplus-0.13.0, xdist-3.3.1, openfiles-0.5.0, cov-4.1.0, astropy-0.10.0, remotedata-0.4.0, asdf-2.15.1, hypothesis-6.82.3
collected 0 items / 1 error                                                                                               

========================================================= ERRORS ==========================================================
______________________________________________ ERROR collecting test session ______________________________________________
.tox/py311-test-devdeps/lib/python3.11/site-packages/_pytest/config/__init__.py:642: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
.tox/py311-test-devdeps/lib/python3.11/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1126: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1126: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1126: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:940: in exec_module
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
astropy/cosmology/__init__.py:11: in <module>
    from . import realizations, units
astropy/cosmology/realizations.py:27: in <module>
    from . import _io  # Ensure IO methods are registered, to read realizations # noqa: F401
astropy/cosmology/_io/__init__.py:8: in <module>
    from . import cosmology, ecsv, html, latex, mapping, model, row, table, yaml
astropy/cosmology/_io/ecsv.py:9: in <module>
    from .table import from_table, to_table
astropy/cosmology/_io/table.py:11: in <module>
    from .utils import convert_parameter_to_column
astropy/cosmology/_io/utils.py:5: in <module>
    from astropy.modeling import Parameter as ModelParameter
astropy/modeling/__init__.py:10: in <module>
    from . import fitting, models
astropy/modeling/fitting.py:39: in <module>
    from .spline import (
astropy/modeling/spline.py:15: in <module>
    from .core import FittableModel, ModelDefinitionError
astropy/modeling/core.py:29: in <module>
    from astropy.nddata.utils import add_array, extract_array
astropy/nddata/__init__.py:13: in <module>
    from .bitmask import *
astropy/nddata/bitmask.py:22: in <module>
    _MAX_UINT_TYPE = np.maximum_sctype(np.uint)
.tox/py311-test-devdeps/lib/python3.11/site-packages/numpy/core/numerictypes.py:176: in maximum_sctype
    warnings.warn(
E   DeprecationWarning: `maximum_sctype` is deprecated. Use an explicit dtype like int64 or float64 instead. (deprecated in NumPy 2.0)
================================================= short test summary info =================================================
ERROR  - DeprecationWarning: `maximum_sctype` is deprecated. Use an explicit dtype like int64 or float64 instead. (deprecated i...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================== 1 error in 0.37s =====================================================
```

Reply:
We already fixed that, so maybe a rebase would do.

Reply:
I dropped the ignore and rebased and force pushed. FYI.